core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -2512,16 +2500,33 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
     def hasNextAt(index: Int): Boolean =
       index < length
     override protected def mapChunk[B](f: Byte => B): Chunk[B] = {
-      val len   = self.length
-      val array = Array.ofDim[AnyRef](len).asInstanceOf[Array[B]]
-
-      var i = 0
-      while (i < len) {
-        array(i) = f(self(i))
-        i += 1
+      val len = self.length
+      if (len == 0) Empty
+      else {
+        val oldArr = array
+        val head   = f(oldArr(0))
+        var newArr = Array.ofDim[B](len)(Chunk.Tags.fromValue(head))
+        var i      = 1
+        newArr(0) = head
+        while (i < len) {
+          val newVal = f(oldArr(i))
+          try {
+            newArr(i) = newVal
+          } catch {
+            case _: ClassCastException =>
```

Comment:
Why do we need this `try catch`? 🤔

Reply:
So this is to prevent issues in a _very_ edge case like this:

```scala
Chunk(1,2,3).map[Any](x => if (x > 1) x.toString else x)
```

Having said that, `.flatMap` doesn't do this and no one had an issue with this so far 🤔

Reply:
Would your example compile? 🤔

Reply:
Without the explicit `[Any]` type annotation your code would probably emit a warning at the compilation, no? 🤔

Reply:
The issue is not for future cases though, it's for maintaining the same behaviour for existing code _just in case_ someone somewhere does this on purpose